firebase-test/src/documents-rules.test.ts

```diff
@@ -159,6 +159,140 @@ describe("Firestore security rules", () => {
+    });
+    
+    it ("student can read their own history entries", async () => {
+      db = initFirestore(studentAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      await expectReadToSucceed(db, kDocumentHistoryDocPath);
+    });
+    
+    it ("student cannot read their own history entries if no parent", async () => {
+      db = initFirestore(studentAuth);
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      await expectReadToFail(db, kDocumentHistoryDocPath);
+    });
+    
+    it ("teacher can read student history", async () => {
+      db = initFirestore(teacherAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      await expectReadToSucceed(db, kDocumentHistoryDocPath);
+    });
+    
+    it ("users authed from different portals cannot read each other's history entries", async () => {
+      db = initFirestore(studentAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      await expectReadToFail(db, "authed/otherPortal/documents/myDocument/history/myHistoryEntry");
+    });
+
+    it ("user cannot read someone else's history entries", async () => {
+      db = initFirestore(genericAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      expectReadToFail(db, kDocumentHistoryDocPath);
+    });
+    
+    it("unauthed user cannot write. Parent doc does not exist", async () => {
+      db = initFirestore();
+      await expectWriteToFail(db, kDocumentHistoryDocPath, specHistoryEntryDoc());
+    });
+    
+    it("unauthed user cannot write. Parent doc exists", async () => {
+      db = initFirestore();
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc());
+      await expectWriteToFail(db, kDocumentHistoryDocPath, specHistoryEntryDoc());
+    });
+    
+    it ("student can write their own history entries", async () => {
+      db = initFirestore(studentAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await expectWriteToSucceed(db, kDocumentHistoryDocPath, specHistoryEntryDoc());
+    });
+
+    it ("users authed from different portals cannot write each other's history entries", async () => {
+      db = initFirestore(studentAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await expectWriteToFail(db, "authed/otherPortal/documents/myDocument/history/myHistoryEntry", specHistoryEntryDoc());
+    });
```

Comment:
I think this write if failing not because of the other portal, but because there is no parent document in the other portal. In this case making an expected to fail test might be worth it:
```suggestion
    it.failing ("users authed from different portals cannot write each other's history entries", async () => {
      db = initFirestore(studentAuth);
      await adminWriteDoc("authed/otherPortal/documents/myDocument", specHistoryEntryParentDoc({add:{uid: studentId }}));
      await expectWriteToFail(db, "authed/otherPortal/documents/myDocument/history/myHistoryEntry", specHistoryEntryDoc());
    });
```